core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/internal/Activator.java

```diff
@@ -88,4 +95,21 @@ public void stop(BundleContext bundleContext) throws Exception {
 
     }
 
+    /**
+     * Set each HashProviderFactory to the HashProviderFactory collection.
+     *
+     * @param bundleContext The OSGi bundle context.
+     * @throws InvalidSyntaxException The exception thrown when the service reference cannot be obtained.
```

Comment:
Better to throw, user core defined exception by handling the `InvalidSyntaxException` within the method. 

As you are using a fixed & valid filter, practically this exception will not be occur in the runtime.
http://docs.osgi.org/javadoc/r4v43/core/org/osgi/framework/BundleContext.html#getServiceReferences(java.lang.Class,%20java.lang.String)

Reply:
fixed with 71869a70460fa550fb5a58bb9fc5bbfc0a104000